src/util/taproot.rs

```diff
@@ -90,6 +90,12 @@ impl TapTweakHash {
         }
         TapTweakHash::from_engine(eng)
     }
+
+    /// Converts a `TapTweakHash` into a `Scalar` ready for use with key tweaking API.
+    pub fn to_scalar(&self) -> Scalar {
+        // This is statistically extremely unlikely to panic (1/2^128).
```

Comment:
Note that wile you're correct it's unlikely, the probability is not `1/2^128` but `432420386565659656852420866394968145599 / 2^256`; still same order of magnitude. :)

Reply:
I copied that from @sanket1729 and do not know where either of these numbers come from, can someone please justify their number so I can understand and I'll fix the comment.

Reply:
For now I have just removed the probability.

Reply:
The number is the difference between our group's size and 2^256, I believe.

Reply:
Isn't the group size equal to p?
```
p = 2^256 - 2^32 - 2^9 - 2^8 - 2^7 - 2^6 - 2^4 - 1
  = 4294968237
```

ref: https://en.bitcoin.it/wiki/Secp256k1

Reply:
It's equal to `n` from the same page. Yes, it's confusing but there are actually two groups/fields in secp256k1 - one for coordinates, the other for points/scalars. Unsurprisingly, this is not the first time I've seen people being confused. :)

(Writing [my own *toy* secp256k1 lib](https://github.com/Kixunil/toy-secp256k1) was a great learning experience, 100% recommend!)

Reply:
> one for coordinates, the other for points/scalars

I'm totally lost by this statement, added writing a toy secp lib to my todo list, good idea!